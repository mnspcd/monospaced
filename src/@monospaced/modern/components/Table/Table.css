.mdrn .Table {
  border-collapse: collapse;
  font-family: var(--font-monospaced);
  font-size: calc(1em * var(--type-scale-s));
  font-variant: slashed-zero;
  line-height: var(--leading);
  width: 100%;
}

.mdrn .Table tbody tr:nth-child(even) {
  background-color: rgba(var(--color-grey-9-rgb), var(--opacity-static));
}

.mdrn .Table--onDarkBackground tbody tr:nth-child(even) {
  background-color: rgba(
    var(--color-white-rgb),
    var(--opacity-static-inverted)
  );
}

.mdrn .Table td {
  font-weight: 300;
}

.mdrn .Table td,
.mdrn .Table th {
  border: calc(1rem / 16) solid var(--color-grey-3);
  height: calc(36rem / 16);
  padding-left: calc(15rem / 16);
  padding-right: calc(15rem / 16);
  vertical-align: middle;
}

.mdrn .Table--onDarkBackground td,
.mdrn .Table--onDarkBackground th {
  border: calc(1rem / 16) solid var(--color-grey-7);
}

@media (--larger-viewport) {
  .mdrn .Table td,
  .mdrn .Table th {
    height: calc(45rem / 16);
    padding-left: calc(23rem / 16);
    padding-right: calc(23rem / 16);
  }
}
