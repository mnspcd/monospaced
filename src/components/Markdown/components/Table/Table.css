.Table {
  border-collapse: collapse;
  font-family: var(--font-monospaced);
  font-size: calc(1em * var(--type-scale-s));
  font-variant: slashed-zero;
  line-height: var(--leading);
  width: 100%;
}

.Table tbody tr:nth-child(even) {
  background-color: rgba(var(--mnspcd-color-grey-9-rgb), var(--opacity-static));
}

.Table--onDarkBackground tbody tr:nth-child(even) {
  background-color: rgba(var(--mnspcd-color-white-rgb), var(--opacity-static-inverted));
}

.Table td {
  font-weight: 300;
}

.Table td,
.Table th {
  border: calc(1rem / 16) solid var(--mnspcd-color-grey-3);
  height: calc(36rem / 16);
  padding-left: calc(15rem / 16);
  padding-right: calc(15rem / 16);
  vertical-align: middle;
}

.Table--onDarkBackground td,
.Table--onDarkBackground th {
  border: calc(1rem / 16) solid var(--mnspcd-color-grey-8);
}

@media (--larger-viewport) {
  .Table td,
  .Table th {
    height: calc(45rem / 16);
    padding-left: calc(23rem / 16);
    padding-right: calc(23rem / 16);
  }
}
