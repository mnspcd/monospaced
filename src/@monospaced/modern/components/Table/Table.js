import classNames from "classnames";
import PropTypes from "prop-types";
import React from "react";

import { SurfaceConsumer } from "../Surface";

import "./Table.css";

/**
 * Use `Table` to display tabular data.
 */
const Table = ({ children }) => (
  <SurfaceConsumer>
    {surfaceBackground => (
      <table
        className={classNames({
          Table: true,
          [`Table--onDarkBackground`]: surfaceBackground === "dark",
          [`Table--onLightBlueBackground`]: surfaceBackground === "blue-light",
        })}
      >
        {children}
      </table>
    )}
  </SurfaceConsumer>
);

Table.propTypes = {
  /**
   * Permitted HTML table content to display in the Table.
   */
  children: PropTypes.node.isRequired,
};

export default Table;
