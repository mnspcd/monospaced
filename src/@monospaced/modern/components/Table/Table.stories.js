import { select } from "@storybook/addon-knobs";
import React from "react";

import Surface from "../Surface";

import Table from "./Table";

export default {
  component: Table,
  title: "Components/Table",
};

export const basic = () => (
  <Table>
    <thead>
      <tr>
        <th align="left">Tables</th>
        <th align="center">Are</th>
        <th align="right">Cool</th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td align="left">col 2 is</td>
        <td align="center">centre-aligned</td>
        <td align="right">£12</td>
      </tr>
      <tr>
        <td align="left">col 3 is</td>
        <td align="center">right-aligned</td>
        <td align="right">£1600</td>
      </tr>
      <tr>
        <td align="left">zebra stripes</td>
        <td align="center">are neat</td>
        <td align="right">£1</td>
      </tr>
    </tbody>
  </Table>
);

export const onDarkBackground = () => (
  <Surface backgroundColor="grey-dark">
    <div style={{ padding: "1em" }}>
      <Table>
        <thead>
          <tr>
            <th align="left">Tables</th>
            <th align="center">Are</th>
            <th align="right">Cool</th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td align="left">col 2 is</td>
            <td align="center">centre-aligned</td>
            <td align="right">£12</td>
          </tr>
          <tr>
            <td align="left">col 3 is</td>
            <td align="center">right-aligned</td>
            <td align="right">£1600</td>
          </tr>
          <tr>
            <td align="left">zebra stripes</td>
            <td align="center">are neat</td>
            <td align="right">£1</td>
          </tr>
        </tbody>
      </Table>
    </div>
  </Surface>
);

export const knobs = () => (
  <Surface
    backgroundColor={select(
      "backgroundColor",
      Surface.backgroundColors,
      "white",
      "Surface",
    )}
  >
    <Table>
      <thead>
        <tr>
          <th align="left">Tables</th>
          <th align="center">Are</th>
          <th align="right">Cool</th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td align="left">col 2 is</td>
          <td align="center">centre-aligned</td>
          <td align="right">£12</td>
        </tr>
        <tr>
          <td align="left">col 3 is</td>
          <td align="center">right-aligned</td>
          <td align="right">£1600</td>
        </tr>
        <tr>
          <td align="left">zebra stripes</td>
          <td align="center">are neat</td>
          <td align="right">£1</td>
        </tr>
      </tbody>
    </Table>
  </Surface>
);

knobs.story = { parameters: { docs: { disable: true } } };
