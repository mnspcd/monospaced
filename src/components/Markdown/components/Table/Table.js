import PropTypes from "prop-types";
import React from "react";
import classNames from "classnames";

import { SurfaceConsumer } from "../../../Surface";
import "./Table.css";

const Table = ({ children }) => (
  <SurfaceConsumer>
    {surfaceBackground => (
      <table
        className={classNames({
          Table: true,
          [`Table--onDarkBackground`]: surfaceBackground === "dark",
          [`Table--onLightBlueBackground`]: surfaceBackground === "blue-light",
        })}
      >
        {children}
      </table>
    )}
  </SurfaceConsumer>
);

Table.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Table;
