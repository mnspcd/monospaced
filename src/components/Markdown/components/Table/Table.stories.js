import React from "react";
import { select } from "@storybook/addon-knobs";
import { storiesOf } from "@storybook/react";

import Surface from "../../../Surface";
import Table from "./Table";

storiesOf("Components/Markdown/Table", module).add("Component", () => (
  <Surface
    backgroundColor={select(
      "backgroundColor",
      Surface.backgroundColors,
      "white",
      "Surface",
    )}
  >
    <Table>
      <thead>
        <tr>
          <th align="left">Tables</th>
          <th align="center">Are</th>
          <th align="right">Cool</th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td align="left">col 2 is</td>
          <td align="center">centre-aligned</td>
          <td align="right">£12</td>
        </tr>
        <tr>
          <td align="left">col 3 is</td>
          <td align="center">right-aligned</td>
          <td align="right">£1600</td>
        </tr>
        <tr>
          <td align="left">zebra stripes</td>
          <td align="center">are neat</td>
          <td align="right">£1</td>
        </tr>
      </tbody>
    </Table>
  </Surface>
));
