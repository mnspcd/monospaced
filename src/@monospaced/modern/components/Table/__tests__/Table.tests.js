/* eslint-env jest */
import { render } from "@testing-library/react";
import React from "react";

import Table from "..";

describe("Table component", () => {
  it("should render correctly", () => {
    const { container } = render(
      <Table>
        <tbody />
      </Table>,
    );

    expect(container).toMatchSnapshot();
  });
});
