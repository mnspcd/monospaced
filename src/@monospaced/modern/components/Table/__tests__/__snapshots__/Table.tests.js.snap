// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table component should render correctly 1`] = `
<div>
  <table
    class="Table"
  >
    <tbody />
  </table>
</div>
`;
